drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -550,6 +555,39 @@ def ce_referral(id:)
       Hmis::Ce::Referral.viewable_by(current_user).find_by(id: id)
     end
 
+    field :direct_referral_form, Types::Forms::FormDefinition, null: true do
+      argument :target_unit_group_id, ID, required: true
+      argument :source_enrollment_id, ID, required: true
+    end
+    def direct_referral_form(target_unit_group_id:, source_enrollment_id:)
+      access_denied! unless Hmis::Ce.configuration.enabled?
+
+      source_enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find(source_enrollment_id)
+      access_denied! unless current_permission?(permission: :can_manage_outgoing_referrals, entity: source_enrollment.project)
+
+      unit_group = Hmis::UnitGroup.find(target_unit_group_id)
+      target_project = unit_group.project # does not need to be viewable by current user
+      access_denied! unless target_project.accepts_direct_ce_referrals_from?(source_enrollment.project)
+
+      workflow_template = unit_group.workflow_template
+
+      user_facing_error_message = "Unit group #{unit_group.name} at project #{target_project.project_name} does not have a correctly configured referral workflow."
+      base_error_message = "UnitGroup:#{unit_group.id} TargetProject:#{target_project.id} SourceEnrollment:#{source_enrollment.id}"
+
+      unless workflow_template&.published? && workflow_template.template_type.to_s == 'ce_referral'
+        error_message = "Workflow template not found. #{base_error_message}"
+        raise HmisErrors::ApiError.new(error_message, display_message: user_facing_error_message)
+      end
+
+      initiation_node = workflow_template.graph.nodes.find(&:delegated_handoff)
+      unless initiation_node&.user_task? && initiation_node.form_definition.present?
+        error_message = "Workflow template #{workflow_template.id} does not have a UserTask node with a form definition that is marked 'delegated_handoff'. #{base_error_message}"
+        raise HmisErrors::ApiError.new(error_message, display_message: user_facing_error_message)
```

Comment:
@martha can you say more about making these user-facing error messages? Can these scenarios arise? The user is unable to do anything about it, so it seems to me it would be preferable to prevent this scenario rather than allowing it with a user-facing error.